compiler/rustc_metadata/src/rmeta/decoder.rs

```diff
@@ -73,6 +73,9 @@ impl MetadataBlob {
 /// own crate numbers.
 pub(crate) type CrateNumMap = IndexVec<CrateNum, CrateNum>;
 
+/// Target modifiers - abi / vulnerability-resist affecting flags
+pub(crate) type TargetModifiers = Vec<String>;
```

Comment:
Can this be a `(String, String)` and we always keep the value separated from the flag?

Reply:
I think it would be good to have a bitset or enum or something that identifies the target modifiers and then put those in this, `(TargetModifier, String)`, so that we can avoid storing so many strings into the crate metadata (which can have a perf impact).

Reply:
The hash is better. I'd like it if we could only have one `value` though, ideally you could encode the actual type of the value here and then when decoding, match on which option it is to know how to decode the value - that would give us the smallest addition to the metadata. As long as a option's value type, like `CodeModel`, is able to be converted back to `small` or the actual thing that a user would write, that should work.